hypothesis-python/src/hypothesis/internal/scrutineer.py

```diff
@@ -38,11 +39,45 @@ def trace(self, frame, event, arg):
         if event == "call":
             return self.trace
         elif event == "line":
-            fname = frame.f_code.co_filename
-            if should_trace_file(fname):
-                current_location = (fname, frame.f_lineno)
-                self.branches.add((self._previous_location, current_location))
-                self._previous_location = current_location
+            self.trace_line(frame.f_code, frame.f_lineno)
+
+    def trace_line(self, code: types.CodeType, line_number: int):
```

Comment:
I think that for `sys.monitoring`, we actually want to `trace_branch` - we're using lines in the existing tracer because branches aren't natively available.

However, this will also require some adjustment of the reporting and maybe analysis logic, so I suggest we get this working with line-monitoring first to get the big performance win, and then refactor to branch analysis in a follow-up.

Reply:
Sounds good. I can take that on after this PR! (unless you want to adjust it yourself).